django_tenants/postgresql_backend/base.py

```diff
@@ -86,14 +86,33 @@ def set_tenant(self, tenant, include_public=True):
             EXTRA_SET_TENANT_METHOD(self, tenant)
 
         self.search_path_set_schemas = None
-
-        # Content type can no longer be cached as public and tenant schemas
-        # have different models. If someone wants to change this, the cache
-        # needs to be separated between public and shared schemas. If this
-        # cache isn't cleared, this can cause permission problems. For example,
-        # on public, a particular model has id 14, but on the tenants it has
-        # the id 15. if 14 is cached instead of 15, the permissions for the
-        # wrong model will be fetched.
+        
+        if has_multi_type_tenants():
+            TENANT_TYPES = getattr(settings, 'TENANT_TYPES')
```

Comment:
Default should be an empty list/dict ?, otherwise if this setting isn't defined iterating the for loop below will crash.

Reply:
Further more getattr will raise AttributeError if default isn't provided.

Reply:
Ah thanks!